app/human-detector-app/App.tsx

```diff
@@ -2,30 +2,75 @@ import * as React from 'react';
 import { View, Text, StyleSheet, Platform } from 'react-native';
 import { NavigationContainer } from '@react-navigation/native';
 import { createNativeStackNavigator } from '@react-navigation/native-stack';
+import { TokenResponse } from 'expo-auth-session';
+import JWT from 'expo-jwt';
+import * as CryptoJS from 'crypto-js';
+import User from './classes/User';
 import CameraScreen from './screens/CameraScreen';
 import GroupScreen from './screens/GroupScreen';
 import LoginScreen from './screens/LoginScreen';
+import { getExponentPushToken, sendExpoNotifToken } from './src/notifications/notifTokenInit';
 
 const Stack = createNativeStackNavigator();
 
 export default function App(): React.ReactElement {
+  let isUserSignedIn = false;
+
+  // Update upon getting a token
+  const [tokenResponse, setTokenResponse] = React.useState(null);
+
+  // Check if the token is expired
+  if (tokenResponse != null) {
+    // TODO: refresh token if expired
+
+    // if not expired, then user is signed in
+    isUserSignedIn = true;
+  }
+
+  // If the user isn't logged in
+  if (!isUserSignedIn) {
```

Comment:
Could this just be an `else` for the previous if statement, and you could get rid of isUserSignedIn? I'm not exactly sure how stuff is persisted between sessions, so not sure if that'd work.

Reply:
The reason why there wont be an `else` here is because the token could be an old token that has an unsuccessful refresh.  It would need to go back to go back to the login screen at that point.  Atleast that's as much as I understand about the flow